src/nnet3/nnet-simple-component.h

```diff
@@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
   Component *CollapseWithNext(const FixedAffineComponent &next) const;
```

Comment:
@danpovey 
A clean-up of the unused code from Affine and other components, to ensure that the unused methods are not copied into newer classes, might be necessary.

Also there is no test for the Vectorize/Unvectorize methods. The Convolution*Components don't implement these methods and there was never a failed test.